xarray/core/variable.py

```diff
@@ -123,6 +123,11 @@ def as_variable(
     if isinstance(obj, Variable):
         obj = obj.copy(deep=False)
     elif isinstance(obj, tuple):
+        try:
+            dims_, data_, *attrs = obj
+        except ValueError:
+            raise ValueError(f"Tuple {obj} is not in the form (dims, data[, attrs])")
+
         if isinstance(obj[1], DataArray):
```

Comment:
```suggestion
        if isinstance(data_, DataArray):
```

I believe this is what @Illviljan is suggesting.

Reply:
Yes! Replace all the `obj` usage downstream as it isn't needed anymore.

Reply:
@dcherian, @Illviljan: Fixed, can you check please?